test/extensions/retry/host/omit_hosts/config_test.cc

```diff
@@ -0,0 +1,86 @@
+#include "envoy/extensions/retry/host/omit_host_metadata/v3alpha/omit_host_metadata_config.pb.h"
+#include "envoy/registry/registry.h"
+#include "envoy/upstream/retry.h"
+
+#include "extensions/retry/host/omit_host_metadata/omit_host_metadata.h"
+#include "extensions/retry/host/well_known_names.h"
+
+#include "test/mocks/upstream/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using namespace testing;
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+namespace {
+
+TEST(OmitHostsRetryPredicateTest, PredicateTest) {
+  auto factory = Registry::FactoryRegistry<Upstream::RetryHostPredicateFactory>::getFactory(
+      RetryHostPredicateValues::get().OmitHostMetadataPredicate);
+
+  ASSERT_NE(nullptr, factory);
+
+  envoy::extensions::retry::host::omit_host_metadata::v3alpha::OmitHostMetadataConfig config;
+  auto* metadata_match = config.mutable_metadata_match();
+  Envoy::Config::Metadata::mutableMetadataValue(
+      *metadata_match, Envoy::Config::MetadataFilters::get().ENVOY_LB, "key")
+      .set_string_value("value");
+  auto empty = factory->createEmptyConfigProto();
+  empty->MergeFrom(config);
+  auto predicate = factory->createHostPredicate(*empty, 3);
+
+  auto host1 = std::make_shared<NiceMock<Upstream::MockHost>>();
```

Comment:
i would just use one host to test with here, unless having separate test is helpful somehow?

Reply:
Done.